metricbeat/module/windows/service/service_status.go

```diff
@@ -232,7 +233,7 @@ func getServiceInformation(rawService *EnumServiceStatusProcess, servicesBuffer
 	if ServiceState(rawService.ServiceStatusProcess.DwCurrentState) != ServiceStopped {
 		processUpTime, err := getServiceUptime(rawService.ServiceStatusProcess.DwProcessId)
 		if err != nil {
-			if _, ok := protectedServices[service.ServiceName]; errors.Cause(err) == syscall.ERROR_ACCESS_DENIED && !ok {
+			if _, ok := protectedServices[service.ServiceName]; errors.Is(err, os.ErrPermission) && !ok {
```

Comment:
Some (but not all) syscall values can just be compared to the higher level error in the `os` or `fs` package: https://pkg.go.dev/syscall#Errno

The ones I did in this way I confirmed this would work by reading the implementation, in this case see line 147 of https://go.dev/src/syscall/syscall_windows.go:

```go
func (e Errno) Is(target error) bool {
	switch target {
	case oserror.ErrPermission:
		return e == ERROR_ACCESS_DENIED
	case oserror.ErrExist:
		return e == ERROR_ALREADY_EXISTS ||
			e == ERROR_DIR_NOT_EMPTY ||
			e == ERROR_FILE_EXISTS
	case oserror.ErrNotExist:
		return e == ERROR_FILE_NOT_FOUND ||
			e == _ERROR_BAD_NETPATH ||
			e == ERROR_PATH_NOT_FOUND
	}
	return false
}
```